app/controllers/tweet_controller.rb

```diff
@@ -0,0 +1,44 @@
+class TweetController < ApplicationController
+  get '/tweets' do
+    redirect '/login' unless session.key?(:user_id)
+    @tweets = Tweet.all
+    @username = User.find(session[:user_id]).username
+    erb :'tweets/index'
+  end
+
+  get '/tweets/new' do
+    redirect '/login' unless session.key?(:user_id)
+    erb :'tweets/new'
+  end
+
+  post '/tweets' do
+    redirect '/login' unless session.key?(:user_id)
+    redirect '/tweets/new' if params[:content].empty?
+    Tweet.create(content: params[:content], user_id: session[:user_id])
+  end
+
+  get '/tweets/:id' do
+    redirect '/login' unless session.key?(:user_id)
+    @tweet = Tweet.find(params[:id])
+    erb :'tweets/show'
+  end
+
+  get '/tweets/:id/edit' do
+    redirect '/login' unless session.key?(:user_id)
+    @tweet = Tweet.find(params[:id])
+    erb :'tweets/edit'
+  end
+
+  patch '/tweets/:id' do
```

Comment:
I would contend that you still want to double check the user's association to the tweet here. There are other ways to submit a PATCH/POST request than going through an HTML form...